public/app/features/panel/metrics_panel_ctrl.ts

```diff
@@ -248,23 +248,27 @@ class MetricsPanelCtrl extends PanelCtrl {
     }
   }
 
-  getAdditionalMenuItems() {
+  async getAdditionalMenuItems() {
     const items = [];
     if (this.contextSrv.hasAccessToExplore() && this.datasource) {
       items.push({
         text: 'Explore',
-        click: 'ctrl.explore();',
+        click: 'ctrl.explore($event);',
         icon: 'gicon gicon-explore',
         shortcut: 'x',
+        href: await getExploreUrl(this.panel.targets, this.datasource, this.datasourceSrv, this.timeSrv),
       });
     }
     return items;
   }
 
-  async explore() {
+  async explore(e: any) {
```

Comment:
this action should not be needed if you have the href?

Reply:
Exactly! 🙂 Based on your message above regarding reloading, we don't need explore function at all, as href will be used.